PlusContent/WFPSupport.md

```diff
@@ -0,0 +1,78 @@
+
+### User Mode Packet Filter Implementation
+
+**Sandboxie Plus v0.9.3** also added a fully functional rule-based packet filter in user mode for the case when **NetworkEnableWFP=y** is not set. This mechanism also replaces the primitive **"BlockPort=..."** functionality of older versions.
+
+**Limitations of the user mode filter:**
+If WFP support is not enabled, the same rules can still be set and used, but they will be applied only by means of user mode hooks. Unlike the WFP implementation, they will **apply only to outgoing connections** and there are no enforcement guarantees as user mode hooks can be bypassed or disabled by a malicious application.
+
+**Caveat:** For reliable isolation, **the use of kernel mode WFP-based filtering is strongly recommended**.
+
+**The Rationale for Two Filtering Modes:**
+The rationale for implementing network functionality in both user mode and kernel mode (driver) is twofold. First, it allows for easier debugging of the rule processing code (simpler to debug in user mode) as both modes use the same code to make decisions based on the preset rules.
+Second, the WFP callouts are global i.e. they are triggered for any process on the system whether sandboxed or not. In the latter case they don't do anything and the use of a hash map to identify sandboxed programs that require action can provide optimal performance.
+Users of a 3rd party firewall may not feel comfortable turning their firewall off and they can continue to use it and still benefit from the selective "per sandbox" network rules offered by Sandboxie Plus. For example, Box1 may allow network access for Program1, while in Box2 the same Program1 may be blocked or even allowed but with a different set of rules for network access.
+
+**Combining WFP with User Mode Filtering:**
+If you set "block internet access" for a given process and have the driver (for WFP) enabled, you can select for that box which method to apply: using WFP or blocking network devices. Even though the approach of blocking the network device endpoints is more absolute, it has been known to cause some applications to crash.
+
+### WFP and Multiple Firewalls
+
+Windows Filtering Platform (WFP) is configured by installing a **provider** of filter rules. <br>
+WFP based firewalls may use the standard Windows Firewall's provider or create their own. <br>
+Some firewalls use WFP at the user mode level (no drivers), while others use WFP in kernel mode (based on their own driver). If several kernel mode firewalls are installed and active at the same time, each driver installs its own callout functions at the positions in the network stack it wants to control and all those functions are then called by the kernel for ALL the drivers (providers). This results in an amalgamation of rules set by each firewall. An inbuilt arbitration mechanism in WFP then decides which rules take precedence.
```

Comment:
About lines 36-37-38, could you find a way to better integrate them into the paragraph as a whole?

I wouldn't know how to do it, but I think it should be done.

Reply:
I see your point! Does the attached file (with your advice and some reshuffling) make a little more sense about WFP firewalls, how they work  (this is not generally well explained) and why Sandboxie can be used simultaneously with one's favorite firewall?
[WFPSupport.md](https://github.com/sandboxie-plus/sandboxie-docs/files/11843440/WFPSupport.md)